### PR TITLE
Join League Functionality

### PR DESCRIPTION
**Pull request by JScheire**
Join league is now fully functional. It will send alerts for joining a public/private league that you are already a part of and also for invalid league keys. I added some other stuff to help with this PR: I added a linter command so that you can run the linter and catch any issues with `npm run lint` and I also turned off `jsx-one-expression-per-line` so I can do stuff like `You successfully joined {selectedLeague.leagueName}!` on one line (line 88, JoinLeague.js). See screens below.

image
image
image
image

### CODE REVIEW

**Review by kjprajapati18 — DISMISSED** [review]

**Review by avidlaud — APPROVED** [review]

File: frontend/src/components/JoinLeague/JoinLeague.js [hunk 1]
```diff
@@ -1,106 +1,150 @@
...
+                        </thead>
+                        <tbody>
+                            {leagues && leagues.map((league) => (
+                                <tr>
+                                    <td>{league.leagueName}</td>
+                                    <td>
+                                        {`${new Date(league.settings.endDate).getMonth() + 1
+                                        }/${new Date(league.settings.endDate).getDate() + 1}/${new Date(league.settings.endDate).getFullYear()}`}
+                                    </td>
+                                    <td>{league.playerList.length}</td>
```
Comment (kjprajapati18) [inline frontend/src/components/JoinLeague/JoinLeague.js:116]: Can you actually make this

<td>{league.playerList.length} / {league.settings.maxPlayers}</td>

for clarity

Comment (kjprajapati18) [reply frontend/src/components/JoinLeague/JoinLeague.js:116]: for line 116

**Review by ag1489 — APPROVED** [review]

**Review by yatri-git — APPROVED** [review]

File: frontend/src/components/JoinLeague/JoinLeague.js (hunk 1 above)
Comment (JScheire) [reply frontend/src/components/JoinLeague/JoinLeague.js]: done

**Comment by avidlaud** [comment]
Is this ready for review?

**Review by kjprajapati18 — DISMISSED** [review]

**Comment by kjprajapati18** [comment]
Just Fixed merge conflicts. Accepted all current changes to packagelock since i don't think it matters. Merge conflict in eslintrc.json was just a new line difference

**Review by kjprajapati18 — APPROVED** [review]

**Review by RiyaTayal20 — APPROVED** [review]

**Review by kjprajapati18 — APPROVED** [review]

**Review by kjprajapati18 — APPROVED** [review]

**Review by avidlaud — APPROVED** [review]

**Pull request merged**